### ISSUE TITLE
checksum feels arbitrary

### ISSUE THREAD

**Issue by pkieltyka** [kardianos/govendor#98]
@kardianos just wondering if the new checksum field in "fetch" branch is really necessary? .. its pretty random. I understand it has something to do with sync to check if a package has changed, but the way its computed and the value feel pretty random

**Comment by kardianos** [comment]
Thank you for looking into the "fetch" branch, I really appreciate it. (Feel free to ping me on slack too.)

The checksum method is as follows:
 * Start at the root directory
 * Sort the listings of the package
 * Add the import path
 * For each file
   - Add file name (important if someone renamed a file to start with a "_" or change suffix)
   - Add file content
 * Go into each sub-folder

The benefits of writing down some type of "checksum" is that `govendor sync` can efficiently determine if the content all there or modified in any way. That way, if you are just testing out a change in a vendor package you can; `govendor status` and `govendor sync` will notice the difference and report it to the developer.

The most important aspect of govendor (as I see it) is introspection. Thus `govendor status` and `govendor list` are probably the most important commands. Without some sort of checksum, I don't see how the `status` command is possible.

Does this make more sense? Do you have further questions?

**Comment by pkieltyka** [comment]
hrmm, yea I understand the reason and figured it had to do with how packages are synced. But then vendor/vendor.json would never be expected to be manipulated by hand, it should be considered as a "lock"ed manifest that should just be changed by the govendor tool, or another lib that will compute all the values. Would you agree? if that is the intention, and everyone is cool with it, then it sounds fine. 

tl;dr: adding checksum and origin fields, I don't see anyone ever updating this file by hand.

**Comment by kardianos** [comment]
The vendor spec is designed to be, use what you need. So in this case, govendor will still work alright if if it doesn't have a checksum field, it will just work a bit more if "sync" is called.

However, practically speaking I do treat this as a "lock" manifest that shouldn't be edited by hand. I'd rather people just use a tool to determine things.

**Comment by pkieltyka** [comment]
@kardianos yea, I think a "lock" manifest that should be managed by govendor tool makes sense. So then, "checksum" field is optional, this all makes sense. Good stuff man. Keep going, we love it and I'm pumping it to everyone I know :)
